sdk/search/azure-search-documents/src/main/java/com/azure/search/annotation/FieldProperty.java

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.annotation;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Retention(RUNTIME)
+@Target({FIELD})
+public @interface FieldProperty {
```

Comment:
I thought we had agreed on using attribute names and properties that match the convenience fields, e.g. `SimpleField`, `SearchableField`, and `ComplexField`.